app/google_calendar/google_calendar_api.py

```
@@ -0,0 +1,243 @@
+            logger.info(f"- summary: {event_result['summary']}")
+            logger.info(f"- starts at: {event_result['start']['dateTime']}")
+            logger.info(f"- ends at: {event_result['end']['dateTime']}")
+            logger.info(f"- location: {event_result['location']}")
+            logger.info(f"- description: {event_result['description']}")
+            logger.info(f"- extendedProperties: {event_result['extendedProperties']}")
+            return True
+
+        except Exception as e:
+            logger.error(f"Failed to update event {event_id} in {self.calendar_id}: {e}")
+            return False
+
+    def list_events(
+        self,
+        fetch_after_this_time: str,
+        max_events_to_fetch: int = settings.NUMBER_OF_EVENTS_TO_FETCH,
+    ) -> list[CalendarEvent]:
+        """List upcoming events in the calendar."""
+
+        logger.info(f"Getting all the events starting after time: {fetch_after_this_time}")
+        calendar_events: list[CalendarEvent] = []
+        try:
+            events_result = (
+                self.service.events()  # type: ignore
+                .list(
+                    calendarId=self.calendar_id,
+                    timeMin=fetch_after_this_time,
+                    singleEvents=True,
+                    maxResults=max_events_to_fetch,
+                    orderBy="startTime",
+                )
+                .execute()
+            )
+            events = events_result.get("items", [])
+            if not events:
+                logger.info(f"No upcoming events found in {self.calendar_id}.")
+            else:
+                logger.info(f"Found {len(events)} upcoming events")
+                # Convert events to CalendarEvent objects
+                calendar_events = [CalendarEvent(**event) for event in events]
+
+            return calendar_events
+
+        except Exception as e:
+            logger.error(f"Failed to list events in {self.calendar_id}: {e}")
```

Why we need to catch the exception here and return empty array? why not just propagate the exception? (for all exceptions)

---

updated